### ISSUE TITLE
[ISSUE] AlarmActivity NullPointerException [Crashlytics]

### ISSUE THREAD

**Issue by davcpas1234** [VersCreativeUK/FACETS#97]
**Application Version**

0.0.7

**Issue Description**

Crashlytics Report

**Stack Trace**
```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String uk.ac.bournemouth.facets.Models.RestPeriod.getRestPeriodToneUri()' on a null object reference
       at uk.ac.bournemouth.facets.Activities.AlarmActivity.playAlarm(AlarmActivity.java:163)
       at uk.ac.bournemouth.facets.Activities.AlarmActivity.access$400(AlarmActivity.java:36)
       at uk.ac.bournemouth.facets.Activities.AlarmActivity$1.onComplete(AlarmActivity.java:144)
       at com.google.android.gms.tasks.zzj.run(:4)
       at android.os.Handler.handleCallback(Handler.java:789)
       at android.os.Handler.dispatchMessage(Handler.java:98)
       at android.os.Looper.loop(Looper.java:164)
       at android.app.ActivityThread.main(ActivityThread.java:6944)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
```

**Comment by davcpas1234** [comment]
Initiating Code:

```
    private void playAlarm(String uri) {
        mediaPlayer = new MediaPlayer();

        Uri toneUri;

        if(TextUtils.isEmpty(uri)) {
            toneUri = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_ALARM);
        }else {
            toneUri = Uri.parse(restPeriod.getRestPeriodToneUri()); // <-- Error Line 163
        }

        try {
            mediaPlayer.setDataSource(this, toneUri);
            mediaPlayer.setAudioStreamType(AudioManager.STREAM_ALARM);
            mediaPlayer.setLooping(true);
            mediaPlayer.prepare();
            mediaPlayer.start();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
```

**Comment by davcpas1234** [comment]
This is due to the code using global variable instead of the passed variable. If the calling code is a SleepWake Alarm `restPeriod` will be undefined (null).

New Code:
```
if(TextUtils.isEmpty(uri)) {
    toneUri = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_ALARM);
} else {
    toneUri = Uri.parse(uri);
}
```

Fix to be release in `v0.0.11-rc1`
